vault/logical_raw.go

```diff
@@ -110,11 +141,83 @@ func (b *RawBackend) handleRawWrite(ctx context.Context, req *logical.Request, d
 		}
 	}
 
-	value := data.Get("value").(string)
+	v := data.Get("value").(string)
+	value := []byte(v)
+	if encoding == "base64" {
+		var err error
+		value, err = base64.StdEncoding.DecodeString(v)
+		if err != nil {
+			return logical.ErrorResponse(err.Error()), logical.ErrInvalidRequest
+		}
+	}
+
+	if req.Operation == UpdateCapability {
+		// Check if this is an existing value with compression applied, if so, use the same compression (or no compression)
+		entry, err := b.barrier.Get(ctx, path)
+		if err != nil {
+			return handleErrorNoReadOnlyForward(err)
+		}
+		if entry == nil {
+			err := fmt.Sprintf("cannot figure out compression type because entry does not exist")
```

Comment:
Do we want to use compression_type if provided in this case?

Reply:
In this case, I think error is better.  This is an update operation and somehow the unexpected occurred (there's no data to update).

Reply:
Right, well there is a race inherent, since we don't lock between when the ExistenceCheck runs and this code runs.  But I'm ok with leaving it as-is.